making-mods/advanced-js/mutation-observers.md

```diff
@@ -0,0 +1,199 @@
+  - [Fullscreen observer](#fullscreen-observer)
+  - [Address bar / mail bar observer](#address-bar--mail-bar-observer)
+  - [Toggling UI element (status bar) observer](#toggling-ui-element-status-bar-observer)
+
+## Basics of mutation observers
+
+The main purpose of Mutation Observers is, as the name suggests, to observe for mutations. These mutations, or changes, can be for things like adding or removing an element and changing attributes like classes on an element.
+
+The basic structure of a Mutation Observer should look something like this:
+
+```JavaScript
+// Fill in "..." with appropriate values
+
+const targetElement = document.querySelector(" ... ");
+const configurationOptions = { ... : true};
+
+function callBack (mutations) {
+  mutations.forEach(function (mutation) {
+    if (mutation.type === ... ) {
+      // act on mutation here
+    }
+  }
+}
+
+const observer = new MutationObserver(callBack);
+observer.observe(targetElement, configurationOptions);
+```
+
+The actual creation of the observer is only contained in the last 2 lines, but in order to successfully get it to do what you want, you need to set up certain values prior to creating the observer (you can define them in place, but it is better practice to define the values beforehand). The 3 parts you need to provide are as follows:
+
+- The target element
+- The configuration options
+- The handler function for mutations
+
+### Target element (`targetElement`)
+
+The target element is the part of the HTML structure that the observer will watch. The element can be selected with methods like: `document.querySelector`, `document.getElementById`, or `document.getElementsByClassname` (when paired with selecting the first result with `[0]`).
+
+### Configuration options (`configurationOptions`)
+
+The configuration options are a JavaScript `object` that sets up the observer to watch for the appropriate types of mutations on the target element. There are many different options, but the 2 most important to mods are `attributes` and `childList`. A more complete explanation of all the options can be found [here](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver/observe#parameters).
+
+- Setting `configurationOptions = {attributes: true}` will set the observer to watch for changes to the attributes of the `targetElement`.
+
+  - This is particularly useful for detecting changes in state represented by adding or removing a class on an element. Like with classes such as `.active` or `.hidden`.
+  - The `#browser` div in Vivaldi contains several state based classes that reflect settings the user has selected.
+
+- Setting `configurationOptions = {childList: true}` will set the observer to watch for child elements of the `targetElement` that are added or removed.
+
+  - Sometimes elements are not hidden or transformed into something else when a change to the UI is made. In these cases, new elements are added, so an observer of this type is useful to detect those changes.
```

Comment:
Isn’t very clear. You say that elms (not the trees 😜) aren’t hidden or transformed, & that they are added, but not that they’re removed.

Not so related to this line, you could note somewhere that you have to watch the parent of the elm being removed.